consensus/src/sys_txn_provider.rs

```diff
@@ -1,7 +1,8 @@
 // Copyright © Aptos Foundation
 
 use aptos_types::system_txn::SystemTransaction;
+use std::sync::Arc;
 
 pub trait SysTxnProvider: Send + Sync {
-    fn get(&self) -> Option<&SystemTransaction>;
+    fn get(&self) -> Option<Arc<SystemTransaction>>;
```

Comment:
So `get` should take/remove the `SystemTransaction` from `SysTxnProvider`? Should we call it `take`?

Reply:
this operation should be read-only, no? The oracle managers should keep proposing until they see the updates are committed.

Reply:
I see. So oracle manager will be responsible to remove committed txns from `SysTxnProvider`.

Reply:
maybe add such a removal interface as well?

Reply:
That can depend on oracle implementation. Let's add it as part of the 1st oracle?